src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
Another reason for use the current implementation is to avoid breaking change as @Eilon suggest